lian/src/lian_nav2/src/main_node.cpp

```diff
@@ -0,0 +1,29 @@
+#include "rclcpp/rclcpp.hpp"
+#include "std_msgs/msg/string.hpp"
+#include "nav_msgs/msg/OccupancyGrid.hpp"
+#include "map.cpp"
+using namespace std::chrono_literals;
+
+class Lian_node : public rclcpp::Node
+{
+public:
+    Lian_node() : Node("lian_node")
+    {
+        map_subscriber_ = this->create_subscription<nav_msgs::msg::OccupancyGrid>(
```

Comment:
You need to create a member to hold the map object
Just create a private member Map map_; 
and use that member in the subscribe